src/main/java/com/hivemq/HiveMQServer.java

```diff
@@ -192,25 +190,50 @@ public static void main(final @NotNull String[] args) throws Exception {
         }
 
         if (shutdownHooks.isShuttingDown()) {
-            return;
+            throw new StartAbortedException();
         }
 
         /* It's important that we are modifying the log levels after Guice is initialized,
         otherwise this somehow interferes with Singleton creation */
         LoggingBootstrap.addLoglevelModifiers();
-        instance.start(null);
+        instance.start(embeddedExtension);
+    }
 
-        if (shutdownHooks.isShuttingDown()) {
+    public void afterStart() {
+        if (injector == null) {
             return;
         }
-
-        log.info("Started HiveMQ in {}ms", TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime));
-
+        final ShutdownHooks shutdownHooks = injector.getInstance(ShutdownHooks.class);
         if (shutdownHooks.isShuttingDown()) {
-            return;
+            throw new StartAbortedException();
         }
 
         final UsageStatistics usageStatistics = injector.getInstance(UsageStatistics.class);
         usageStatistics.start();
     }
+
+    public void startFully() throws Exception {
```

Comment:
pls change the naming like suggested in the EE PR

Reply:
done